test/unit/SES/Transport/SESTransportTest.php

```diff
@@ -0,0 +1,228 @@
+use Swift_Mime_SimpleMessage;
+use test\mock\Ingenerator\SwiftMailer\SES\Transport\SpyingSESClient;
+
+class DummySwiftPlugin implements Swift_Events_SendListener
+{
+    protected array $got_events;
+
+    public function assertGotEvents(array $expect)
+    {
+        Assert::assertSame($expect, $this->got_events);
+    }
+
+    public function beforeSendPerformed(Swift_Events_SendEvent $evt)
+    {
+        $this->got_events[] = [__FUNCTION__, $evt];
+    }
+
+    public function sendPerformed(Swift_Events_SendEvent $evt)
+    {
+        $this->got_events[] = [__FUNCTION__, $evt];
+    }
+
+}
+
+class DummySwift_Message extends Swift_Message
+{
+    private string $dummy_mime_string;
+
+    public static function returnDummyMimeString(string $string): DummySwift_Message
+    {
+        $s                    = new self();
+        $s->dummy_mime_string = $string;
+
+        return $s;
+    }
+
+    public function toString(): string
+    {
+        return $this->dummy_mime_string;
+    }
+}
+
+class SESTransportTest extends TestCase
+{
+    protected Swift_Events_EventDispatcher $event_dispatcher;
+
+    protected SpyingSESClient $ses_client;
+
+    protected ArrayMetricsAgent $metrics;
+
+    protected TestLogger $log;
+
+    public function test_it_is_initialisable()
+    {
+        $this->assertInstanceOf(SESTransport::class, $this->newSubject());
+    }
+
+    public function test_it_is_started()
+    {
+        $this->assertTrue($this->newSubject()->isStarted());
+    }
+
+    public function test_it_throws_with_more_than_one_from_address()
+    {
+        $message = new Swift_Message;
+        $message->addFrom('foo@bar.com');
+        $message->addFrom('other@bar.com');
+        $subject = $this->newSubject();
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('multiple `from` address');
+        $subject->send($message);
+    }
+
+    public function test_it_throws_with_more_than_one_to_address()
+    {
+        $message = new Swift_Message;
+        $message->addTo('foo@bar.com');
+        $message->addTo('other@bar.com');
+        $subject = $this->newSubject();
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('multiple `to` address');
+        $subject->send($message);
+    }
+
+    public function test_its_register_plugin_just_binds_to_event_listener()
+    {
+        $plugin  = new DummySwiftPlugin;
+        $subject = $this->newSubject();
+        $subject->registerPlugin($plugin);
+
+        $evt = $this->event_dispatcher->createSendEvent($subject, $this->getDummy(Swift_Mime_SimpleMessage::class));
+        $this->event_dispatcher->dispatchEvent($evt, 'sendPerformed');
+        $plugin->assertGotEvents([['sendPerformed', $evt]]);
+    }
+
+    public function test_its_send_method_sends()
+    {
+        $subject = $this->newSubject();
+        $message = $this->generateValidMessage();
+        $subject->send($message);
```

Comment:
`$this->assertTrue($subject->send($message))` just to cover the return value as well?